### PR TITLE
feat: add optional trace to CairoVM struct and add enable trace flag

### LINKED ISSUE #9: feat: add optional trace to CairoVM struct and add enable trace flag
Right now the CairoVM is [missing an optional trace field](https://github.com/keep-starknet-strange/cairo-zig/blob/main/src/vm/core.zig#L14), which when we are in `enable-trace` mode should be updated with every step of the virtual machine. The trace is a vector of `TraceEntry` of the registers `pc`, `ap`, and `fp`. In `runInstruction` `[function](https://github.com/keep-starknet-strange/cairo-zig/blob/main/src/vm/core.zig#L104)` we need to push a new `TraceEntry` struct that receives the registers from `RunContext` [register holder](https://github.com/keep-starknet-strange/cairo-zig/blob/main/src/vm/run_context.zig#L12). 

It's OK to push a default value of `RunContext` for the moment but when [this](https://github.com/keep-starknet-strange/cairo-zig/issues/10) is complete the actual value needs to be pushed. 

Add the `enable-trace` flag in [cmd](https://github.com/keep-starknet-strange/cairo-zig/blob/main/src/cmd/cmd.zig).

### PR DESCRIPTION
**Pull request by itzlambda**
fix: #9

### CODE REVIEW

**Review by AbdelStark — CHANGES_REQUESTED** [review]
left comment

File: src/vm/core.zig [hunk 1]
```diff
@@ -439,7 +471,11 @@ test "update pc regular with imm" {
     instruction.op_1_addr = instructions.Op1Src.Imm;
     const operands = OperandsResult.default();
     // Create a new VM instance.
-    var vm = try CairoVM.init(&allocator);
+    var config = Config{
```
Comment (abdelhamidbakhta) [inline src/vm/core.zig:474]: can you add a `default` function in `Config` and use it in tests ?

Comment (lambda-0x) [reply src/vm/core.zig:474]: right thats much better

Comment (bingcicle) [reply src/vm/core.zig]: As far as I recall you can also set defaults in the struct itself here:

https://github.com/keep-starknet-strange/cairo-zig/blob/3abd1bdfae155497ec9f46c8ff88f1b0b46c2183/src/vm/config.zig#L6-L8

Like so:

```zig
pub const Config = struct { 
     proof_mode: bool = false, 
     enable_trace: bool = false, 
```

and pass `.{}` everywhere:
```zig
var vm = try CairoVM.init(&allocator, .{});
```

Comment (lambda-0x) [reply src/vm/core.zig]: interesting!

**Comment by lambda-0x** [comment]
haha lets try again: #29

**Pull request closed without merging**